frontend/js/pages/BattleDetail.js

```diff
@@ -13,34 +13,46 @@ function BattleDetail(props) {
   const { id } = useParams();
   const { battle } = props.battle;
   const { user } = props.user;
-
+  const fetchedBattle = props.location.query;
   useEffect(() => {
     if (!user) {
       props.getCurrentUser();
     }
-    props.getBattle(id);
+    if (!fetchedBattle) {
+      props.getBattle(id);
+    }
   }, []);
 
-  if (!battle) {
+  if (!battle && !fetchedBattle) {
     return '';
   }
-
-  const { current, other } = orderTeamsByCurrentUser(battle, user);
-
+  let pokemonsList = null;
+  const battleDetail = fetchedBattle ? fetchedBattle.battle : battle.entities.battle[id];
+  const trainerDetail = fetchedBattle ? fetchedBattle.battleUsers : battle.entities.user;
+  if (fetchedBattle) {
+    pokemonsList = fetchedBattle.pokemons ? fetchedBattle.pokemons : null;
+  } else if (battle.entities) {
+    pokemonsList = battle.entities.pokemon ? battle.entities.pokemon : null;
+  }
+  const { current, other } = orderTeamsByCurrentUser(battleDetail, user, trainerDetail);
```

Comment:
This logic is not 100%, but the update to be made here is to move all the logic to `mapStateToProps` and use denormalize to get the data in the right format for the component